trino/trino_test.go

```diff
@@ -765,6 +765,106 @@ func TestQueryColumns(t *testing.T) {
 	assert.Equal(t, actualTypes, expectedTypes)
 }
 
+func TestParametricDateTime(t *testing.T) {
+	c := &Config{
+		ServerURI:         *integrationServerFlag,
+		SessionProperties: map[string]string{"query_priority": "1"},
+	}
+
+	dsn, err := c.FormatDSN()
+	require.NoError(t, err)
+
+	db, err := sql.Open("trino", dsn)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		assert.NoError(t, db.Close())
+	})
+
+	rows, err := db.Query(`SELECT
+  cast(current_time as time(6)) AS timep,
+  cast(current_time as time(6) with time zone) AS timeptz,
+  cast(current_time as timestamp(6)) AS tsp,
+  cast(current_time as timestamp(6) with time zone) AS tsptz`,
+		sql.Named("X-Trino-Client-Capabilities", "PARAMETRIC_DATETIME"))
```

Comment:
Can this be enabled by default? This would simplify usage and reduce cognitive load - I don't remember how exactly this parameter behaves and would have to check Trino sources or other clients. I wouldn't expect users of this driver to know this.

Whatever we decide here, the README also should be updated.

Reply:
In every client which supports this we want it to be the default behaviour - the Python client will start supporting this in next release and also defaults to always use the PARAMETRIC_DATETIME behaviour. It's a "breaking change" in some sense but IMO the benefits of always using parametric_datetime outweigh possible breakage.

Reply:
I have no problem enabling this since I set it in my own code, let me know if it's OK and if there should be a mechanism to not pass it. I stumbled upon this while investigating why my timestamps were truncated to milliseconds: https://github.com/trinodb/trino/issues/5068

Reply:
In Python client we bit the bullet and enabled it by default with no option to fallback (since I can't think of any case where the alternative behaviour of precision always being 3 is useful).

I'd like to hear @losipiuk's opinion as well on this.

Reply:
Enabled by default, let me know.

Reply:
README updated to document the date/time support